src/NetherPortalScanner.h

```diff
@@ -48,11 +48,14 @@ class cNetherPortalScanner : public cChunkStay
 	/** Whether the given location is a valid location to build a portal. */
 	bool IsValidBuildLocation(Vector3i a_BlockPosition);
 
-	/** The entity that's being moved. */
-	cEntity * m_Entity;
+	/** The ID of the entity that's being moved. */
+	UInt32 m_EntityID;
+
+	/** The world we're moving the entity from, used to query the entity ID. */
+	cWorld & m_SourceWorld;
 
 	/** The world we're moving the entity to. */
-	cWorld * m_World;
+	cWorld & m_World;
```

Comment:
This may be academic but I've heard that having references as members can lead to undefined behaviour. As a general rule, I try to avoid them. Using references for the constructor is good though.

A more practical point. Should we worry about `m_SourceWorld` becoming invalidated when the server shuts down?

Reply:
Interesting! I changed them here because indeed the portal scanner is inseparably bound to a world so I thought it would make this dependency clearer.

I'm fairly certain shutting down is a whole can of worms in itself, but given that the scanner is driven by the worlds' tick threads it shouldn't run code when the world ends.

Reply:
You're referencing the _other_ world. There could be some tick where one world is running but the other is already shutdown.

Reply:
The order should be, all worlds are created, all tick threads start, (server running), (server asked to stop), all tick threads stop, all worlds deleted.

This should preclude any chance of one world still ticking trying to access a dead world.

Reply:
And then you get someone asking "But I want to create a new world while the server is running", and once you have that, "But I want to remove a world while the server is running" :P

Reply:
darn users always asking for stuff

Anyway I'm only describing how it is right now.

Reply:
>all tick threads stop, all worlds deleted

Okay, I verified that's true and there is synchronisation before deleting. So we should be good.